### ISSUE TITLE
Cannot read properties of undefined (reading 'number')

### ISSUE THREAD

**Issue by stevelacey** [peter-evans/create-pull-request#2790]
### Subject of the issue

I started getting this error on v4, so upgraded to v6, but I'm still seeing it 🤔

```
Create or update the pull request
  Attempting creation of pull request
  A pull request already exists for example:branch
  Fetching existing pull request
  Attempting update of pull request
  Error: Cannot read properties of undefined (reading 'number')
```

Seems to only happen when there's an existing pull request, if I delete the existing one the workflow will succeed.

### Steps to reproduce

Run the workflow with an existing pull request present.

```
    - uses: peter-evans/create-pull-request@v6
      with:
        title: Release ${{ inputs.next_version }}
        body: |
          ## Release [${{ inputs.next_version }}](${{ github.server_url }}/${{ github.repository }}/compare/${{ inputs.previous_version }}...${{ steps.commit.outputs.short }})

          ${{ inputs.notes }}
        author: GitHub <noreply@github.com>
        branch: release
        committer: GitHub <noreply@github.com>
        commit-message: |
          chore(release): ${{ inputs.next_version }}
        delete-branch: true
```

https://github.com/stevelacey/release-builder/blob/v1/pull-request/action.yml#L27C25-L39

**Comment by walterholohan** [comment]
Same here, starting happening a few hours ago

**Comment by peter-evans** [comment]
I've not made any changes to the action since releasing `v6`, so this looks like an issue with GitHub's API.

Reproduced the problem with my test suite.
https://github.com/peter-evans/create-pull-request-tests/actions/runs/8071848530

I'll make a GitHub support ticket.

**Comment by peter-evans** [comment]
Created a GitHub support ticket for this issue. Thanks for reporting it.

**Comment by peter-evans** [comment]
I'm trying to confirm with GitHub support, but it looks like there was a bug with GitHub's list pull requests API that just got fixed. It appears I was unknowingly working around this bug for about 2 years! In fixing it they've caused a breaking change.

**Comment by peter-evans** [comment]
Released `v6.0.1` / `v6`.

Please note that I'm assuming that GitHub are not going to roll this change back. 😓 I'm trying to confirm with GitHub support that it was actually a long standing bug that just got fixed.

**Comment by peter-evans** [comment]
GitHub support have confirmed that an intentional change was made to the list API, and the fix they suggested to me (released as `v6.0.1` / `v6`) should be good from now on. Not expecting this change to be rolled back.

**Comment by peter-evans** [comment]
Due to another issue preventing some users from being able to use `v6`, this fix is now backported to `v5` in `v5.0.3`.
